src/main/java/epam/gym/dao/hibernateImpl/TraineeDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+package epam.gym.dao.hibernateImpl;
+
+import epam.gym.dao.TraineeDao;
+import epam.gym.entity.Trainee;
+import epam.gym.entity.Trainer;
+import epam.gym.entity.Training;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Repository
+public class TraineeDaoImpl implements TraineeDao {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
+    @Transactional
+    public Trainee create(Trainee trainee) {
+        entityManager.persist(trainee);
+        return trainee;
+    }
+
+    @Override
+    @Transactional
+    public Trainee update(Trainee trainee) {
+        return entityManager.merge(trainee);
+    }
+
+    @Override
+    @Transactional
```

Comment:
In most cases, @Transactional is not required for read operations

Reply:
deleted from read methods